uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/models/instances/leave-request-instance.js

```diff
@@ -247,9 +269,53 @@ define([
          * @param {string} key - The property name
          */
         toAPIFilter: function (result, __, key) {
-          if (!_.includes(['balance_change', 'dates', 'comments'], key)) {
+          if (!_.includes(['balance_change', 'dates', 'comments', 'uploader', 'files'], key)) {
             result[key] = this[key];
           }
+        },
+
+        /**
+         * Loads file attachments associated with this leave request
+         *
+         * @return {Promise}
+         */
+        loadAttachments: function () {
+          return LeaveRequestAPI.getAttachments(this.id)
+            .then(function (attachments) {
+              this.files = attachments;
+            }.bind(this));
+        },
+
+        /**
+         * Sets the flag to mark file for deletion. The file is not yet deleted
+         * from the server.
+         */
+        deleteAttachment: function (attachmentID) {
```

Comment:
The internal details of instance like delete flag etc., need not be exposed to controller. It should simply call the method on instance itself to mark it for deletion. what say?